src/api/product/product.controller.ts

```diff
@@ -3,13 +3,29 @@ import { Request, Response } from 'express'
 import {
   getAllProducts,
   createProduct,
-  updateProduct
+  updateProduct,
+  getProductById
 } from './product.service'
 
+export async function getProductByIdHandler(req: Request, res: Response) {
+  const { id } = req.params
+  try {
+    const product = await getProductById(id)
+    return res.status(200).json(product)
+
+  } catch (error: any) {
+    console.log(error)
```

Comment:
also this one.

Reply:
misma respuesta anterior Hector :) esto es intencional